### PR TITLE
Issue 302: Change ConstructorDeclarations throws clause to have Refer…

### PR DESCRIPTION
**Pull request by matozoid**
…enceTypes instead of NameExprs.

@ftomassetti - I need you to check this carefully, since I simply copied your change here: https://github.com/javaparser/javaparser/commit/13ce95600c7c6badc1a65962ad21bc34152b7318

### CODE REVIEW

File: javaparser-testing/src/test/resources/com/github/javaparser/bdd/visitor_scenarios.story [hunk 1]
```diff
@@ -73,4 +73,4 @@ Given a CompilationUnit
 Given a VoidVisitorAdapter with a visit method that asserts sensible line positions
 When the "JavaConcepts.java" is parsed
 When the CompilationUnit is visited by the PositionTestVisitor
-Then the total number of nodes visited is 1425
+Then the total number of nodes visited is 1427
```
Comment (ftomassetti) [inline javaparser-testing/src/test/resources/com/github/javaparser/bdd/visitor_scenarios.story:76]: This test is a bit annoying to change every time. Is it still a good idea to have it?

File: javaparser-core/src/main/javacc/java_1_8.jj [hunk 2]
```diff
@@ -1711,11 +1711,13 @@ ConstructorDeclaration ConstructorDeclaration(Modifier modifier):
 	int bbColumn = 0;
 	int beLine = 0;
 	int beColumn = 0;
+	Type throwType;
 }
 {
   [ typeParameters = TypeParameters() { int[] lineCol=(int[])typeParameters.remove(0); if(line==-1){ line=lineCol[0]; column=lineCol[1];} } ]
   // Modifiers matched in the caller
-  name = SimpleName() { if(line==-1){line=token.beginLine; column=token.beginColumn;}} parameters = FormalParameters() [ "throws" throws_ = NameList() ]
+  name = SimpleName() { if(line==-1){line=token.beginLine; column=token.beginColumn;}} parameters = FormalParameters() [ "throws" throwType = ReferenceTypeWithAnnotations() { throws_ = add(throws_, throwType); }
```
Comment (ftomassetti) [inline javaparser-core/src/main/javacc/java_1_8.jj:1719]: I cannot remember how JavaCC works on this so I am going to trust you here :)

**Comment by ftomassetti** [comment]
It seems good to me. I wonder why I made the change just for methods and not also for constructors :)

**Comment by matozoid** [comment]
Because the original PR turned into a long discussion ;)

**Pull request merged**